casper/src/main/scala/coop/rchain/casper/Estimator.scala

```diff
@@ -50,83 +52,44 @@ object Estimator {
+                                currScore       = acc.getOrElse(hash, 0)
+                                validatorWeight <- weightFromValidator[F](b, validator)
+                              } yield acc.updated(hash, currScore + validatorWeight)
+                          }
+      } yield updatedScoreMap
 
-    //add scores to the blocks implicitly supported through
-    //including a latest block as a "step parent"
-    def addValidatorWeightToImplicitlySupported(scoreMap: Map[BlockHash, Int],
-                                                internalMap: Map[BlockHash, BlockMessage],
-                                                childMap: Map[BlockHash, Set[BlockHash]],
-                                                validator: Validator,
-                                                latestBlockHash: BlockHash) =
-      childMap
-        .get(latestBlockHash)
-        .foldLeft(scoreMap) {
-          case (acc, children) =>
-            children.filter(scoreMap.contains).foldLeft(acc) {
-              case (acc, cHash) =>
-                val c = internalMap(cHash)
-                if (parents(c).size > 1 && c.sender != validator) {
-                  val currScore       = acc.getOrElse(cHash, 0)
-                  val validatorWeight = weightMap(c).getOrElse(validator, 0)
-                  acc.updated(cHash, currScore + validatorWeight)
-                } else {
-                  acc
-                }
-            }
-        }
-
-    def addValidatorWeightToBlockScore(acc: Map[BlockHash, Int],
-                                       internalMap: Map[BlockHash, BlockMessage],
-                                       childMap: Map[BlockHash, Set[BlockHash]],
-                                       validator: Validator,
-                                       latestBlockHash: BlockHash) = {
-      val scoreMap =
-        addValidatorWeightDownSupportingChain(acc, internalMap, validator, latestBlockHash)
-      addValidatorWeightToImplicitlySupported(scoreMap,
-                                              internalMap,
-                                              childMap,
-                                              validator,
-                                              latestBlockHash)
-    }
-
-    blockDag.latestMessages.foldLeft(Map.empty[BlockHash, Int]) {
-      case (acc, (validator: Validator, latestBlockHash: BlockHash)) =>
-        addValidatorWeightToBlockScore(acc,
-                                       internalMap,
-                                       blockDag.childMap,
-                                       validator,
-                                       latestBlockHash)
-    }
+    for {
+      scoresMap <- Foldable[List].foldM(blockDag.latestMessages.toList, Map.empty[BlockHash, Int]) {
+                    case (acc, (validator: Validator, latestBlockHash: BlockHash)) =>
+                      addValidatorWeightDownSupportingChain(acc, validator, latestBlockHash)
```

Comment:
I've gotten rid of that as it seems we are heading towards the direction of splitting transaction fees in a stake-weighted fashion. What was the conclusion on that front?

Reply:
I haven't seen the final spec yet, which would include that decision, but I think so. But the implicit support thing was to make the fork choice prefer multi-parent blocks because they are good for the network. I don't see what rewards has to do with that.

Reply:
Done